From 1240cc16d1b6466a692404f88965bbea883ed3c4 Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Thu, 6 Jul 2017 23:26:19 -0700
Subject: [PATCH 2534/2748] HADOOP-14511.
 WritableRpcEngine.Invocation#toString NPE on null
 parameters. Contributed by John Zhuge.

(cherry picked from commit 87f4428de017e5bfc4358304728bd2281d501d52)

Change-Id: I62b9d6a2578d00e35d9b0445d40941cc5e7abf98
---
 .../org/apache/hadoop/ipc/WritableRpcEngine.java   |   11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/WritableRpcEngine.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/WritableRpcEngine.java
index 0e3e0b0..688e47a 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/WritableRpcEngine.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/WritableRpcEngine.java
@@ -183,10 +183,13 @@ public String toString() {
       StringBuilder buffer = new StringBuilder();
       buffer.append(methodName);
       buffer.append("(");
-      for (int i = 0; i < parameters.length; i++) {
-        if (i != 0)
-          buffer.append(", ");
-        buffer.append(parameters[i]);
+      if (parameters != null) {
+        for (int i = 0; i < parameters.length; i++) {
+          if (i != 0) {
+            buffer.append(", ");
+          }
+          buffer.append(parameters[i]);
+        }
       }
       buffer.append(")");
       buffer.append(", rpc version="+rpcVersion);
-- 
1.7.9.5

